src/benchmark_mediation.py

```diff
@@ -935,6 +935,248 @@ def medDML(y, t, m, x, trim=0.05, order=1):
+        res = LassoCV(alphas=alphas, cv=CV_FOLDS).fit(xm[train_mean1], y[train_mean1])
+        mu_t1_m_x[i] = res.predict(xm[test])
+        mu_t1_m_x_nested[i] = res.predict(xm[train_nested])
+
+        # predict E[Y|T=0,M,X]
+        res = LassoCV(alphas=alphas, cv=CV_FOLDS).fit(xm[train_mean0], y[train_mean0])
+        mu_t0_m_x[i] = res.predict(xm[test])
+        mu_t0_m_x_nested[i] = res.predict(xm[train_nested])
+
+        # predict E[E[Y|T=1,M,X]|T=0,X]
+        res = LassoCV(alphas=alphas, cv=CV_FOLDS).fit(
+            x[train_nested0], mu_t1_m_x_nested[i][t[train_nested] == 0]
+        )
+        w_t0_x[i] = res.predict(x[test])
+
+        # predict E[E[Y|T=0,M,X]|T=1,X]
+        res = LassoCV(alphas=alphas, cv=CV_FOLDS).fit(
+            x[train_nested1], mu_t0_m_x_nested[i][t[train_nested] == 1]
+        )
+        w_t1_x[i] = res.predict(x[test])
+
+        # predict E[Y|T=1,X]
+        res = LassoCV(alphas=alphas, cv=CV_FOLDS).fit(x[train1], y[train1])
+        mu_t1_x[i] = res.predict(x[test])
+
+        # predict E[Y|T=0,X]
+        res = LassoCV(alphas=alphas, cv=CV_FOLDS).fit(x[train0], y[train0])
+        mu_t0_x[i] = res.predict(x[test])
+
+        # trimming
+        not_trimmed = (
+            (((1 - ptmx[i]) * ptx[i]) >= trim)
+            * ((1 - ptx[i]) >= trim)
+            * (ptx[i] >= trim)
+            * (((ptmx[i] * (1 - ptx[i]))) >= trim)
+        )
+        for var in var_name:
+            exec(f"{var}[i] = {var}[i][not_trimmed]")
+        nobs += np.sum(not_trimmed)
+
+    # score computing
+    if normalized:
+        sumscore1 = [np.mean(_) for _ in (1 - tte) * ptmx / ((1 - ptmx) * ptx)]
+        sumscore2 = [np.mean(_) for _ in tte / ptx]
+        sumscore3 = [np.mean(_) for _ in (1 - tte) / (1 - ptx)]
+        sumscore4 = [np.mean(_) for _ in tte * (1 - ptmx) / (ptmx * (1 - ptx))]
+        y1m1 = (tte * (yte - mu_t1_x) / ptx) / sumscore2 + mu_t1_x
+        y0m0 = ((1 - tte) * (yte - mu_t0_x) / (1 - ptx)) / sumscore3 + mu_t0_x
+        y1m0 = (
+            (tte * (1 - ptmx) / (ptmx * (1 - ptx)) * (yte - mu_t1_m_x)) / sumscore4
+            + ((1 - tte) / (1 - ptx) * (mu_t1_m_x - w_t0_x)) / sumscore3
+            + w_t0_x
+        )
+        y0m1 = (
+            ((1 - tte) * ptmx / ((1 - ptmx) * ptx) * (yte - mu_t0_m_x)) / sumscore1
+            + (tte / ptx * (mu_t0_m_x - w_t1_x)) / sumscore2
+            + w_t1_x
+        )
+    else:
+        y1m1 = tte * (yte - mu_t1_x) / ptx + mu_t1_x
+        y0m0 = (1 - tte) * (yte - mu_t0_x) / (1 - ptx) + mu_t0_x
+        y1m0 = (
+            tte * (1 - ptmx) / (ptmx * (1 - ptx)) * (yte - mu_t1_m_x)
+            + (1 - tte) / (1 - ptx) * (mu_t1_m_x - w_t0_x)
+            + w_t0_x
+        )
+        y0m1 = (
+            (1 - tte) * ptmx / ((1 - ptmx) * ptx) * (yte - mu_t0_m_x)
+            + tte / ptx * (mu_t0_m_x - w_t1_x)
+            + w_t1_x
+        )
+
+    # mean score computing
+    my1m1 = np.mean([np.mean(_) for _ in y1m1])
```

Comment:
Can't you make it simpler ?

Reply:
wdym @bthirion, not having an intermediate step to compute the average, or compute the average without list comprehension? (or something else)?

Reply:
I think that the code is fine and the variables well-names. Long functions are a bit dangerous for maintenance, hence it would be better to modularize the code. 
But this can certainly be done in a later PR.